zip-0226.rst

```diff
@@ -177,21 +177,31 @@ The use of different value base points for different Assets enables the final ba
-:math:`\mathsf{bsk} = \sum_{\mathsf{ \forall i\in \{1,...,n\}}} \mathsf{rcv_{i}}`.
+:math:`\mathsf{bsk} = \sum_{i=1}^{n} \mathsf{rcv_{i}}`.
 
 Then the verifier MUST compute
 
-.. math:: \mathsf{bvk = (\sum cv_i^{net})}  - \mathsf{ ValueCommit_0^{Orchard}(v^{balanceOrchard})} = \sum \mathsf{rcv_{i}^{net}}\mathcal{R}^{\mathsf{Orchard}}
+.. math:: \mathsf{bvk} = \mathsf{(\sum_{i=1}^{n} cv_i^{net})}  - \mathsf{ValueCommit_0^{OrchardZSA}(v^{balanceOrchard}, \mathcal{V}^{\mathsf{Orchard}})}
 
-and use it to verify the `bindingSignature` on the `SIGHASH` message.
+After computing :math:`\mathsf{bvk}`, the verifier MUST use it to verify the `bindingSignature` on the `SIGHASH` message.
 
 Rationale for Value Balance Verification
 ````````````````````````````````````````
 
-The main reason why no changes to the Orchard process are needed is that no Custom Assets can be unshielded, so all Custom Assets are contained within the shielded pool. This means that the net balance of the input and output values is zero, with only one Asset of value balance published, that of ZEC, :math:`\mathsf{v^{balanceOrchard}}`. No net amount of any other Asset will be revealed, and the number of Assets in the transaction is also hidden. The only exception to this is in the case that an Asset is *burnt*, as we will see below in the `burn mechanism`_.
+We assume :math:`n` Actions in a transfer. Out of these :math:`n` Actions, we further distinguish (for the sake of verbosity and clarity) between Actions related to ZEC and Actions related to Custom Assets. We assume :math:`l` Actions related to ZEC and :math:`n-l` actions related to Custom Assets, where :math:`l \in [0,n]`.
+
+The value balance verification is equivalent to:
+
+.. math:: \mathsf{bvk} = \mathsf{(\sum_{i=1}^{l} cv_i^{net}) + (\sum_{j=l+1}^{n} cv_i^{net})} - \mathsf{ValueCommit_0^{OrchardZSA}(v^{balanceOrchard}, \mathcal{V}^{\mathsf{Orchard}})}
+
+.. math:: \mathsf{bvk} = [\mathsf{(\sum_{i=1}^{l} cv_i^{net})} - \mathsf{ValueCommit_0^{OrchardZSA}(v^{balanceOrchard}, \mathcal{V}^{\mathsf{Orchard}})}] + \mathsf{(\sum_{j=l+1}^{n} cv_i^{net})} 
+    
+The first term of this equation gives the balance check of the Orchard protocol [#protocol-binding]_. For a correctly constructed transaction, we get :math:`\mathsf{(\sum_{i=1}^{l} v_i^{net}) = v^{balanceOrchard}}`, which means that :math:`\mathsf{(\sum_{i=1}^{l} cv_i^{net})} - \mathsf{ValueCommit_0^{OrchardZSA}(v^{balanceOrchard}, \mathcal{V}^{\mathsf{Orchard}})}` MUST be equal to :math:`\mathsf{\sum_{i=1}^{l} rcv_{i}^{net}}\mathcal{R}^{\mathsf{Orchard}}`. With ZSA, transfer Actions for Custom Assets MUST also be balanced across asset bases. As such, for a correctly constructed transaction, we MUST get :math:`\mathsf{(\sum_{j=l+1}^{n} v_i^{net}) = 0}`, and thus :math:`\mathsf{\sum_{j=l+1}^{n} rcv_{j}^{net}}\mathcal{R}^{\mathsf{Orchard}}`, where :math:`\sum_{j=l+1}^{n} x_j` gives the identity element of the group, if :math:`l+1 > n`.
```

Comment:
"For a correctly constructed transaction ..." - there is no point in repeating the spec in such detail. Remove this sentence.

Also, what is `x_j` ?

And I don't think that `l+1` can be larger than `n`.

Reply:
> And I don't think that l+1 can be larger than n.

If l = n (i.e. all actions move ZEC), then l+1 > n

> Also, what is x_j ?

An arbitrary group element. This is introduced to tackle the edge case where l=n and thus l+1 > n, so we want to make it clear that the sum returns the identity element in the group (in code, the "for loop condition" would fail and thus the for loop instruction's would not be executed.)